qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -830,7 +830,7 @@ def _get_sweep_data(self, force_polar: bool = False) -> np.ndarray:
                     data_format_command = "SDAT"
                 else:
                     data_format_command = "FDAT"
-                timeout = self.sweep_time() + self._additional_wait
+                timeout = self.sweep_time() * 1.5 + self._additional_wait
```

Comment:
I would suggest that rather than exposing additional_wait that we simply allow the user to overwrite timeout.

This could be done as (handling the sentinel value of timeout of None meaning no timeout)
```
old_timeout = self.root_instrument.timeout()
if old_timeout is None:
    old_timeout = float("+inf")
timeout = max(self.sweep_time() * 1.5,  old_timeout)
```

That way a user can simply overwrite timeout if the calculated time is too short.

Reply:
I would expect that if the timeout is not set, then the default `1.5 * self.sweep_time()` should be used, and zero or negative timeout is `+inf`.

Reply:
Yes but that is not how it is implemented in pyvisa. We follow pyvisa except that the qcodes timeout is in sec not ms.

See https://pyvisa.readthedocs.io/en/latest/introduction/resources.html#timeout

```
Here, my_device may be a device, an interface or whatever, and its timeout is set to 25 seconds. To set an infinite timeout, set it to None or float('+inf') or:

del my_device.timeout
Now every operation of the resource takes as long as it takes, even indefinitely if necessary.

To set it to immediate, set it to 0 or a negative value. (Actually, any value smaller than 1 is considered immediate)

Attributes of MessageBase resources[](https://pyvisa.readthedocs.io/en/latest/introduction/resources.html#attributes-of-messagebase-resources)
```

Reply:
thanks @einsmein Could I get you to add a changelog item for this and then I think we are ready to merge